Source/Core/Core/NetPlayServer.cpp

```diff
@@ -221,11 +225,22 @@ void NetPlayServer::ThreadFunc()
       m_ping_timer.Start();
       SendToClients(spac);
 
-      m_index.SetPlayerCount(static_cast<int>(m_players.size()));
-      m_index.SetGame(m_selected_game);
-      m_index.SetInGame(m_is_running);
+      if (Config::Get(Config::NETPLAY_USE_INDEX) && m_index.HasActiveSession())
+      {
+        m_index.SetPlayerCount(static_cast<int>(m_players.size()));
+        m_index.SetGame(m_selected_game);
+        m_index.SetInGame(m_is_running);
 
-      m_update_pings = false;
+        m_update_pings = false;
+      }
+      else
+      {
+        if (m_dialog != nullptr)
+        {
+          m_dialog->OnIndexRefreshFailed(m_index.GetLastError());
+          Config::SetCurrent(Config::NETPLAY_USE_INDEX, false);
```

Comment:
This probably shouldn't be here. In fact this code path will execute every time it's called if the index is disabled or failed.

Reply:
Not the cause of the massive slowdown though.

Reply:
Yeah I know, that was totally unrelated, but this still spams the chat.